ci/nightly.template.steps.yaml

```
@@ -50,16 +51,23 @@ macos: &macos
   - ssh://git@github.com/improbable/apple-codesigning-plugin#v0.1.3: ~
 
 steps:
-  - label: "build-and-deploy-BUILKDITE_AGENT_PLACEHOLDER-${ENGINE_COMMIT_HASH}"
-    command: ${BUILDKITE_COMMAND}
-    <<: *BUILKDITE_AGENT_PLACEHOLDER 
+  - label: "build-and-deploy-AGENT_PLACEHOLDER-ENGINE_COMMIT_HASH_PLACEHOLDER"
+    command: "COMMAND_PLACEHOLDER"
+    <<: *AGENT_PLACEHOLDER 
+    key: "setup-and-build-AGENT_PLACEHOLDER-ENGINE_COMMIT_FORMATTED_HASH_PLACEHOLDER"
+    depends_on:
+      - "generate-auth_token"
     artifact_paths:
       - "UnrealEngine/Engine/Programs/AutomationTool/Saved/Logs/*"
       - "cooked-android/**/*"
       - "cooked-mac/**/*"
       - "cooked-ios/**/*"
     env:
-      ENGINE_COMMIT_HASH: "${ENGINE_COMMIT_HASH}"
+      ENGINE_COMMIT_HASH: "ENGINE_COMMIT_HASH_PLACEHOLDER"
+      #ENGINE_COMMIT_FORMATTED_HASH is a env variable for subsequent steps
+      #it's very similar to NGINE_COMMIT_HASH, but replace spaces,-,. as _ to distinguish engine_version of UnrealEngine.
```

```suggestion
      # It's very similar to ENGINE_COMMIT_HASH, but replaces ` `, `-`, `.` as `_` to distinguish engine_version of UnrealEngine.
```
I'm not sure I follow the `to distinguish engine_version of UnrealEngine.` part, could you expand the comment please?